x/programs/rust/examples/nft/src/lib.rs

```diff
@@ -159,16 +158,23 @@ pub fn burn(program: Program, from: Address) -> bool {
     // Burn the NFT by transferring it to the zero address
     program
         .state()
-        .store(StateKey::Balance(from).to_vec(), &(balance - BURN_AMOUNT))
+        .store(StateKey::Balances(from).to_vec(), &(balance - BURN_AMOUNT))
         .expect("failed to store new balance");
 
-    // TODO move to a lazy static? Or move to the VM layer entirely
     program
         .state()
-        .store(StateKey::Owner.to_vec(), &null_address)
+        .store(StateKey::Owners(from).to_vec(), &null_address)
```

Comment:
Does this even delete the address from the map? Also, what if the owner has multiple NFTs?